internal/providers/terraform/aws/elasticsearch_domain.go

```diff
@@ -0,0 +1,110 @@
+package aws
+
+import (
+	"github.com/infracost/infracost/pkg/schema"
+	"github.com/shopspring/decimal"
+)
+
+func NewElasticsearchDomain(d *schema.ResourceData, u *schema.ResourceData) *schema.Resource {
+
+	region := d.Get("region").String()
+	clusterConfig := d.Get("cluster_config").Array()[0]
+	instanceType := clusterConfig.Get("instance_type").String()
+	instanceCount := clusterConfig.Get("instance_count").Int()
+	ebsOptions := d.Get("ebs_options").Array()[0]
+
+	ebsTypeMap := map[string]string{
+		"gp2":      "GP2",
+		"io1":      "PIOPS-Storage",
+		"standard": "Magnetic",
+	}
+
+	gbVal := decimal.NewFromInt(int64(defaultVolumeSize))
+	if ebsOptions.Get("volume_size").Exists() {
+		gbVal = decimal.NewFromFloat(ebsOptions.Get("volume_size").Float())
+	}
+
+	ebsType := "gp2"
+	if ebsOptions.Get("volume_type").Exists() {
+		ebsType = ebsOptions.Get("volume_type").String()
+	}
+
+	ebsFilter := "gp2"
+	if val, ok := ebsTypeMap[ebsType]; ok {
+		ebsFilter = val
+	}
+
+	iopsVal := decimal.NewFromInt(1)
+	if ebsOptions.Get("iops").Exists() {
+		iopsVal = decimal.NewFromFloat(ebsOptions.Get("iops").Float())
+
+		if iopsVal.LessThan(decimal.NewFromInt(1)) {
+			iopsVal = decimal.NewFromInt(1)
+		}
+	}
+
+	costComponents := []*schema.CostComponent{
+		{
+			Name:           "Per instance hour",
```

Comment:
"Instance" should be enough as the unit clarifies it and the calculation already includes the count, so I'm thinking it means "The line item is the instance of your cloud resource, that internally has a few nodes/masters/warm nodes etc"